### PR TITLE
46 plugin server

### LINKED ISSUE #46: Plugin Server
- [ ] Azure Storage
- [ ] Plugin Upload System
- [ ] Plugin Index
- [ ] Plugin Download
- [ ] Plugin Manager

### PR DESCRIPTION
**Pull request by g-martin772**
I know this is quite incomplete but due to resource constraints we need to move along and get this merged as is. I hope the cq is still ok.

Closes #46

### CODE REVIEW

**Comment by g-martin772** [comment]
I didn't update the docs yet, are we fine to postpone this to the actual release for the final roundup?

**Comment by NotSayo** [comment]
> I didn't update the docs yet, are we fine to postpone this to the actual release for the final roundup?

Yop, we will need to do that if we wanna even think of actually meeting the deadlines. Leave the docs for now. We won't need them if nothing is working anyway

File: src/controller/plugins/Endpoints/TestEndpoints.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace plugins.Endpoints;
+
+public static class TestEndpoints
+{
```
Comment (NotSayo) [inline src/controller/plugins/Endpoints/TestEndpoints.cs:8]: I assume this is supposed to be here. Honestly let's just let it be...

File: src/server/Server/Data/ServerUser.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+using Microsoft.AspNetCore.Identity;
+
+namespace Server.Data;
+
```
Comment (NotSayo) [inline src/server/Server/Data/ServerUser.cs:4]: Do we really need a seperate user for this? Any chance we can use the SystemUser in the future?

File: src/controller/plugins/Endpoints/TestEndpoints.cs (hunk 1 above)
Comment (g-martin772) [reply src/controller/plugins/Endpoints/TestEndpoints.cs:8]: Hmmm, good question

**Review by NotSayo — APPROVED** [review]
Looks good, I assume it works? Think about ways to integrate the user with the one w have in the backend to avoid conflicts

File: src/server/Server/Data/ServerUser.cs (hunk 2 above)
Comment (g-martin772) [reply src/server/Server/Data/ServerUser.cs:4]: Das is am server, auf azure, für plugin autoren, ganz eigene  welt

**Comment by g-martin772** [comment]
> Looks good, I assume it works? Think about ways to integrate the user with the one w have in the backend to avoid conflicts

As mentioned, these are 2 completely different users. The ServerUser represents a plugin user uploading plugins to azure storage, the SystemUser represents a local user on a per controller basis.

> I assume this is supposed to be here. Honestly let's just let it be...

Probably a remnant of all the file copying around from designing the template, gonna delete it

**Comment by g-martin772** [comment]
Looks like we have an issue with the plugin test pipeline, gonna review that later

**Comment by g-martin772** [comment]
Run fine locally on both windows and linux, i honestly don't care right now

**Pull request merged**